test/trento/application/process_managers/deregistration_process_manager_test.exs

```diff
@@ -74,13 +146,83 @@ defmodule Trento.DeregistrationProcessManagerTest do
     end
   end
 
+  describe "host registration procedure" do
+    test "should add database instance with the proper host id when DatabaseInstanceRegistered event is emitted" do
```

Comment:
I miss a bit the fact the we don't test that the new instance is _added_ to the list. We always start with an empty list, which could somehow hide issues